src/core/annotations/field-value.ts

```diff
@@ -0,0 +1,37 @@
+import {isPresent} from 'angular2/src/facade/lang';
+
+
+declare var Symbol: any;
+
+
+/**
+ * Annotation Factory that allows HTML style boolean attributes. For example,
+ * a field declared like this:
+
+ * @Directive({ selector: 'component' }) class MyComponent {
+ *   @Input() @BooleanFieldValueFactory() myField: boolean;
+ * }
+ *
+ * You could set it up this way:
+ *   <component myField>
+ * or:
+ *   <component myField="">
+ */
+function booleanFieldValueFactory() {
+  return function booleanFieldValueMetadata(target: any, key: string): void {
+    const defaultValue = target[key];
+
+    // Use a fallback if Symbol isn't available.
+    const localKey = isPresent(Symbol) ? Symbol(key) : `@@$_${key}`;
```

Comment:
adding special characters such as `@@` makes the object into slow object. Better to use `__private_symbol_` or something like that.